sdk/python/lib/pulumi/x/automation/workspace.py

```diff
@@ -0,0 +1,350 @@
+        Sets all values in the provided config map for the specified stack name.
+
+        :param stack_name: The name of the stack.
+        :param config: A mapping of key to ConfigValue to set to config.
+        """
+        pass
+
+    @abstractmethod
+    def remove_config(self, stack_name: str, key: str) -> None:
+        """
+        Removes the specified key-value pair on the provided stack name.
+
+        :param stack_name: The name of the stack.
+        :param key: The key to remove from config.
+        """
+        pass
+
+    @abstractmethod
+    def remove_all_config(self, stack_name: str, keys: List[str]) -> None:
+        """
+        Removes all values in the provided key list for the specified stack name.
+
+        :param stack_name: The name of the stack.
+        :param keys: The keys to remove from config.
+        """
+        pass
+
+    @abstractmethod
+    def refresh_config(self, stack_name: str) -> None:
+        """
+        Gets and sets the config map used with the last update for Stack matching stack name.
+
+        :param stack_name: The name of the stack.
+        """
+        pass
+
+    @abstractmethod
+    def who_am_i(self) -> WhoAmIResult:
+        """
+        Returns the currently authenticated user.
+
+        :returns: WhoAmIResult
+        """
+        pass
+
+    @abstractmethod
+    def stack(self) -> Optional[StackSummary]:
+        """
+        Returns a summary of the currently selected stack, if any.
+
+        :returns: Optional[StackSummary]
+        """
+        pass
+
+    @abstractmethod
+    def create_stack(self, stack_name: str) -> None:
+        """
+        Creates and sets a new stack with the stack name, failing if one already exists.
+
+        :param str stack_name: The name of the stack to create
+        :returns: None
+        :raises CommandError Raised if a stack with the same name exists.
+        """
+        pass
+
+    @abstractmethod
+    def select_stack(self, stack_name: str) -> None:
+        """
+        Selects and sets an existing stack matching the stack stack_name, failing if none exists.
+
+        :param stack_name: The name of the stack to select
+        :returns: None
+        :raises CommandError Raised if no matching stack exists.
+        """
+        pass
+
+    @abstractmethod
+    def remove_stack(self, stack_name: str) -> None:
+        """
+        Deletes the stack and all associated configuration and history.
+
+        :param stack_name: The name of the stack to remove
+        """
+        pass
+
+    @abstractmethod
+    def list_stacks(self) -> List[StackSummary]:
+        """
+        Returns all Stacks created under the current Project.
+        This queries underlying backend and may return stacks not present in the Workspace
+        (as Pulumi.<stack>.yaml files).
+
+        :returns: List[StackSummary]
+        """
+        pass
+
+    @abstractmethod
+    def install_plugin(self, plugin_name: str, version: str, kind: str) -> None:
```

Comment:
Yes, but in [python](https://docs.python.org/3/library/typing.html#typing.Optional), `Optional` means that `None` is a valid value for that parameter, which is not the case here. 

> Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional.

That being said, I wasn't sure if the abstract method needed to specify the defaults (it is specified in the [LocalWorkspace](https://github.com/pulumi/pulumi/pull/5979/files#diff-ec008bd37a4735c2e8200ae287b2614e6d6586bddbe38c5b596817aba90ab8b4R324) implementation).